src/main/kotlin/org/koitharu/kotatsu/parsers/site/TruyentranhLHParser.kt

```diff
@@ -0,0 +1,154 @@
+			altTitle = infoEl?.selectFirst(".info-item:contains(Tên khác) > .info-value")?.text(),
+			author = infoEl?.select(".info-item:contains(Tác giả) > .info-value")?.joinToString { it.text() },
+			tags = tags ?: emptySet(),
+			state = state,
+			rating = rating ?: RATING_UNKNOWN,
+			description = infoHeaderEl?.selectFirst(".series-summary .summary-content")?.html(),
+			chapters = docs.select("ul.list-chapters.at-series > a").reversed().mapChapters { index, element ->
+				MangaChapter(
+					id = generateUid(element.attrAsRelativeUrl("href")),
+					name = element.selectFirst(".chapter-name")?.text()?.trim().orEmpty(),
+					number = index + 1,
+					url = element.attrAsRelativeUrl("href"),
+					scanlator = null,
+					uploadDate = chapterDateFormat.tryParse(element.selectFirst(".chapter-time")?.text()),
+					branch = null,
+					source = source,
+				)
+			},
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val sortQuery = when (sortOrder) {
+			SortOrder.UPDATED -> "update"
+			SortOrder.NEWEST -> "new"
+			SortOrder.RATING -> "like"
+			SortOrder.POPULARITY -> "top"
+			SortOrder.ALPHABETICAL -> "az"
+		}
+		val url = urlBuilder().apply {
+			addPathSegment("tim-kiem")
+			addQueryParameter("sort", sortQuery)
+			addQueryParameter("page", page.toString())
+			if (!query.isNullOrEmpty()) {
+				addQueryParameter("q", query)
+			}
+			if (!tags.isNullOrEmpty()) {
+				val tagsQuery = tags.joinToString(separator = ",") { it.key }
+				addEncodedQueryParameter("accept_genres", tagsQuery)
+			}
+		}.build()
+
+		return webClient.httpGet(url).parseHtml()
+			.select(".container .card.card-dark .row > .thumb-item-flow")
+			.mapNotNull {
+				val a = it.selectFirstOrThrow(".thumb-wrapper a")
+				Manga(
+					id = generateUid(a.attrAsRelativeUrl("href")),
+					url = a.attrAsRelativeUrl("href"),
+					publicUrl = a.attrAsAbsoluteUrl("href"),
+					title = it.select(".thumb_attr.series-title").text(),
+					altTitle = null,
+					rating = RATING_UNKNOWN,
+					isNsfw = false,
+					coverUrl = a.selectFirst("div[data-bg]")?.attrAsAbsoluteUrl("data-bg").orEmpty(),
+					tags = emptySet(),
+					state = null,
+					author = null,
+					source = source,
+				)
+			}
+	}
+
+	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+		val url = chapter.url.toAbsoluteUrl(domain)
+		return webClient.httpGet(url).parseHtml().select("#chapter-content > img").mapNotNull {
+			val imageUrl = it.attrAsRelativeUrlOrNull("data-src")
+				?: it.attrAsRelativeUrlOrNull("src")
+				?: return@mapNotNull null
+			MangaPage(
+				id = generateUid(imageUrl),
+				url = imageUrl,
+				referer = url,
+				preview = null,
+				source = source,
+			)
+		}
+	}
+
+	override suspend fun getTags(): Set<MangaTag> {
+		return ArraySet(getOrCreateTagMap().values)
+	}
+
+	private suspend fun getOrCreateTagMap(): Map<String, MangaTag> = mutex.withLock {
```

Comment:
We have a new [`SuspendLazy`](https://github.com/KotatsuApp/kotatsu-parsers/blob/master/src/main/kotlin/org/koitharu/kotatsu/parsers/util/SuspendLazy.kt) class for this cases)

Reply:
Okay, I will use it next time =)